Compound reaction loading

samseaver opened the pull request:
Debugged and expanded the addcpdtable and addrxntable commands.  Of note is the ability to now state which current namespace you would like to match the identifiers to, i.e.:

```
ms bio addcpdtable BioCyc  <file1> -n MetaCyc -m BioCyc
ms bio addcpdtable BioCyc  <file2> -n PlantCyc -m BioCyc
```

Printing out the number of compounds in each namespace gives you:

```
MetaCyc 12358
PlantCyc 3776
BioCyc 12506
```

Showing how roughly 3500 PlantCyc identifiers match MetaCyc identifiers.

samseaver committed:
Added three things to Reaction.pm a) Removal of protons from equation code if using hashed string b) Expanded code to handle a wider variety of characters in the compound identifiers in reaction equations. c) Used a hash to count the number of times a compound occurs in a single compartment within the same equation, and returns a zero/fail if this occurs more than once as it indicates an equation that cannot be used in metabolic modeling."

samseaver committed:
Expanded the addCompoundFromHash and addReactionFromHash functions to do: a) handle long names b) use a mergeto namespace to match identifiers c) report reactions rejected for having duplicate compounds

samseaver committed:
Updated addcpdtable and addrxntable commands to allow for verbosity and to include a mergeto option for indicating the namespace to which identifiers can be compared.  Tested successfully with several BioCyc databases

samseaver committed:
Expanding reaction discovery by matching equations. Reaction id is likely to come from diferent namespace

samseaver committed:
Merge branch 'master' into Compound_Reaction_Loading

samseaver committed:
Expanding report of matching reactions to correctly indicate the database and identifier that the match came from

cshenry commented:
Nice work Sam!

On Oct 16, 2012, at 12:38 AM, samseaver wrote:

> Debugged and expanded the addcpdtable and addrxntable commands. Of note is the ability to now state which current namespace you would like to match the identifiers to, i.e.:
> 
> ms bio addcpdtable BioCyc  <file1> -n MetaCyc -m BioCyc
> ms bio addcpdtable BioCyc  <file2> -n PlantCyc -m BioCyc
> 
> Printing out the number of compounds in each namespace gives you:
> 
> MetaCyc 12358
> PlantCyc 3776
> BioCyc 12506
> 
> Showing how roughly 3500 PlantCyc identifiers match MetaCyc identifiers.
> 
> You can merge this Pull Request by running:
> 
>   git pull https://github.com/ModelSEED/ModelSEED Compound_Reaction_Loading
> Or view, comment on, or merge it at:
> 
>   https://github.com/ModelSEED/ModelSEED/pull/114
> 
> Commit Summary
> 
>   • Added three things to Reaction.pm a) Removal of protons from equation…
>   • Expanded the addCompoundFromHash and addReactionFromHash functions to…
>   • Updated addcpdtable and addrxntable commands to allow for verbosity a…
> File Changes
> 
>   • M lib/ModelSEED/App/bio/Command/addcpdtable.pm (56)
>   • M lib/ModelSEED/App/bio/Command/addrxntable.pm (56)
>   • M lib/ModelSEED/MS/Biochemistry.pm (177)
>   • M lib/ModelSEED/MS/Reaction.pm (42)
> Patch Links
> 
>   • https://github.com/ModelSEED/ModelSEED/pull/114.patch
>   • https://github.com/ModelSEED/ModelSEED/pull/114.diff
> —
> Reply to this email directly or view it on GitHub.

devoid committed:
Fix compile error in Biochemistry

- $arg undefined in ModelSEED::MS::Biochemistry function
  addCompoundFromHash. Replace with correct name `$args`

devoid committed:
Fix same error in reaction

Beware of copying code.
